src/target/stm32f4.c

```diff
@@ -423,9 +423,21 @@ static bool stm32f4_attach(target_s *const target)
 	/* And rebuild the RAM map */
 	bool use_dual_bank = !is_f7 && dual_bank;
 	if (is_f7) {
-		target_add_ram32(target, 0x00000000, 0x4000);  /* 16kiB ITCM RAM */
-		target_add_ram32(target, 0x20000000, 0x20000); /* 128kiB DTCM RAM */
-		target_add_ram32(target, 0x20020000, 0x60000); /* 384kiB RAM */
+		uint32_t dtcm_size = 64U * 1024U; /* 64kiB DTCM RAM */
+		uint32_t ahbsram_size = 16U * 1024U;
+		if (target->part_id == ID_STM32F72X) {
+			ahbsram_size = (176U + 16U) * 1024U; /* 192kiB SRAM1/2 */
+		} else if (target->part_id == ID_STM32F74X) {
+			ahbsram_size = (240U + 16U) * 1024U; /* 256kiB SRAM1/2 */
+		} else if (target->part_id == ID_STM32F76X) {
+			dtcm_size = 128U * 1024U;            /* 128kiB DTCM RAM */
+			ahbsram_size = (368U + 16U) * 1024U; /* 384kiB SRAM1/2 */
+		}
+		target_add_ram32(target, 0x00000000, 0x4000); /* 16kiB ITCM RAM */
+		/* On STM32F7, DTCM and AHB SRAM are contiguous */
+		target_add_ram32(target, 0x20000000, dtcm_size);
+		target_add_ram32(target, 0x20000000 + dtcm_size, ahbsram_size);
```

Comment:
As you are here working on these, please `U` suffix the addresses and size information and turn them into `#define`'s at the top of file as appropriate.

Reply:
Suffixed base addresses in new/edited code. Hoisted `STM32F7_DTCM_BASE` up top. Ignored 0x0U ITCM RAM base for now (`ITCM_BASE` is for the Flash alias, not RAM), and I'm not sure someone needs an alias for address zero.

Reply:
We do as it has specific meaning and is a valid address in the map - given it's the `ITCM_BASE`, please name it as `STM32F7_ITCM_BASE`.

Reply:
Named it `STM32F7_ITCM_RAM_BASE`, extending the other newly added names as well.